src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/SqlServerType.cs

```diff
@@ -0,0 +1,91 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using Microsoft.SqlTools.ServiceLayer.Connection.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Nodes
+{
+    /// <summary>
+    /// Server Types
+    /// </summary>
+    public enum SqlServerType
+    {
+        Unknown,
+        Sql2005,
+        Sql2008,
+        Sql2012,
+        Sql2014,
+        Sql2016,
+        Azure,
+        AzureV12
+    }
+
+    /// <summary>
+    /// Includes helper functions for server version and type
+    /// </summary>
+    public class ServerVersionHelper
+    {
+        /// <summary>
+        /// Converts a server type to ValidForFlag
+        /// </summary>
+        public static ValidForFlag GetValidForFlag(SqlServerType serverType)
+        {
+            ValidForFlag validforFlag = ValidForFlag.All;
+            if (Enum.TryParse<ValidForFlag>(serverType.ToString(), out validforFlag))
+            {
+                return validforFlag;
+            }
+            return ValidForFlag.All;
+        }
+
+        /// <summary>
+        /// Creates a server type from the server version
+        /// </summary>
+        public static SqlServerType CalculateServerType(ServerInfo serverInfo)
+        {
+            SqlServerType serverType = SqlServerType.Unknown;
+            string serverVersion = serverInfo.ServerVersion;
+
+            if (serverInfo.IsCloud)
+            {
+                if (serverVersion.StartsWith("11", StringComparison.Ordinal))
```

Comment:
Good news: Azure v11 is no more! We only have to handle AzureV12 now. This will simplify the logic somewhat. In theory there might be a future version of Azure that's different enough to re-introduce this logic, but I don't see that happening for a while.